app-config/src/secret-agent.ts

```diff
@@ -201,15 +211,23 @@ export function shouldUseSecretAgent(value?: boolean) {
 
 const defaultPort = 42938;
 
-async function getAgentPort(portOverride?: number) {
-  if (portOverride !== undefined) {
-    return portOverride;
+async function getAgentPortOrSocket(
+  socketOrPortOverride?: number | string,
+): Promise<number | string> {
+  if (socketOrPortOverride !== undefined) {
+    return socketOrPortOverride;
   }
+
   const settings = await loadSettingsLazy();
 
+  if (settings.secretAgent?.socket) {
+    return settings.secretAgent.socket;
```

Comment:
loads the `socket` property if present in settings file, otherwise defaults to port. I didn't add a default for socket, so that we can roll it out as default later if we want, and because detecting unix-like environments might be slightly complicated.